cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -543,18 +543,30 @@ public void saveProgramRuntimeArgs(HttpRequest request, HttpResponder responder,
   }
 
   @GET
-  @Path("/apps/{app-id}/{program-type}/{program-id}")
+  @Path("/apps/{app-name}/{program-type}/{program-name}")
   public void programSpecification(HttpRequest request, HttpResponder responder,
-                                   @PathParam("namespace-id") String namespaceId, @PathParam("app-id") String appId,
+                                   @PathParam("namespace-id") String namespaceId, @PathParam("app-name") String appName,
                                    @PathParam("program-type") String type,
-                                   @PathParam("program-id") String programId) throws Exception {
+                                   @PathParam("program-name") String programName) throws Exception {
+    programSpecification(request, responder, namespaceId, appName, ApplicationId.DEFAULT_VERSION, type, programName);
+  }
+
+
+  @GET
+  @Path("/apps/{app-name}/versions/{app-version}/{program-type}/{program-name}")
+  public void programSpecification(HttpRequest request, HttpResponder responder,
```

Comment:
In 3.6 there are still some versioned REST API left to be implemented, and this happens to be necessary for a `ProgramClient` method